src/main/java/org/killbill/billing/plugin/analytics/api/user/AnalyticsUserApi.java

```diff
@@ -114,4 +117,35 @@ public void rebuildAnalyticsForAccount(final UUID accountId, final CallContext c
         allBusinessObjectsDao.update(businessContextFactory);
         logger.info("Finished Analytics refresh for account {}", businessContextFactory.getAccountId());
     }
+
+    public void rebuildAnalyticsForAllAccounts(final CallContext context) {
+        logger.info("Starting Analytics refresh for all accounts");
+        Pagination<Account> accounts = osgiKillbillAPI.getAccountUserApi().getAccounts(0L, 100L, context);
+        rebuildAnalyticsForAccounts(accounts, context);
```

Comment:
I'm using the `http://127.0.0.1:8080/1.0/kb/accounts/pagination` endpoint to get a List of account IDs as specified in the [Readme](https://github.com/killbill/killbill-analytics-plugin). Wondering if we should instead write a method within the `analyticsDao` that returns a list of all the account IDs within the analytics database?

Reply:
> See my comment below: Analytics tables are a mirror of the Kill Bill tables, so we want to use the Kill Bill database (API) as the source of truth.

okay, makes sense 👍 
>BTW, here, you're not using the HTTP endpoint but using the Java API instead (which is the right thing to do, just wanted to make sure you realized

Yup I do, I meant the Java API method, I guess I did not pay much attention while typing this out 😛